pkg/controllers/deprovisioning/controller.go

```diff
@@ -110,17 +110,20 @@ func (c *Controller) Reconcile(ctx context.Context, _ reconcile.Request) (reconc
 	// with making any scheduling decision off of our state nodes. Otherwise, we have the potential to make
 	// a scheduling decision based on a smaller subset of nodes in our cluster state than actually exist.
 	if !c.cluster.Synced(ctx) {
+		logging.FromContext(ctx).Debugf("waiting for in-memory cluster state to sync before deprovisioning")
 		return reconcile.Result{RequeueAfter: time.Second}, nil
 	}
 	// Attempt different deprovisioning methods. We'll only let one method perform an action
 	isConsolidated := c.cluster.Consolidated()
 	for _, d := range c.deprovisioners {
+		ctx = context.WithValue(ctx, "deprovisioner", d.String())
 		candidates, err := GetCandidates(ctx, c.cluster, c.kubeClient, c.clock, c.cloudProvider, d.ShouldDeprovision)
 		if err != nil {
 			return reconcile.Result{}, fmt.Errorf("determining candidates, %w", err)
 		}
 		// If there are no candidate nodes, move to the next deprovisioner
 		if len(candidates) == 0 {
+			logging.FromContext(ctx).With("deprovisioner", ctx.Value("deprovisioner")).Debugf("found no deprovisionable nodes")
 			continue
 		}
 
```

Comment:
What if the `ComputeCommand` below just hangs indefinitely?  I think we would just see deprovisioning stop, with no indication of why.  If we had a log per deprovisioner, we would know if one of them hung.

Reply:
I'm confused. If the compute command hangs indefinitely, we wouldn't see this line. Isn't that the point?